yt_dlp/extractor/archiveorg.py

```diff
@@ -698,18 +891,21 @@ def _real_extract(self, url):
         url_date = url_date or url_date_2
 
         urlh = None
-        try:
-            urlh = self._request_webpage(
-                HEADRequest('https://web.archive.org/web/2oe_/http://wayback-fakeurl.archive.org/yt/%s' % video_id),
-                video_id, note='Fetching archived video file url', expected_status=True)
-        except ExtractorError as e:
-            # HTTP Error 404 is expected if the video is not saved.
-            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
-                self.raise_no_formats(
-                    'The requested video is not archived, indexed, or there is an issue with web.archive.org',
-                    expected=True)
-            else:
-                raise
+        retry_manager = self.RetryManager(fatal=False)
+        for retry in retry_manager:
+            try:
+                urlh = self._request_webpage(
+                    HEADRequest('https://web.archive.org/web/2oe_/http://wayback-fakeurl.archive.org/yt/%s' % video_id),
+                    video_id, note='Fetching archived video file url', expected_status=True)
+            except ExtractorError as e:
+                retry.error = e
+                # HTTP Error 404 is expected if the video is not saved.
+                if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
+                    retry.error = ExtractorError(
+                        'The requested video is not archived, indexed, or there is an issue with web.archive.org', expected=True)
+
+        if retry_manager.error:
+            self.raise_no_formats(retry_manager.error, expected=True, video_id=video_id)
```

Comment:
RetryManager fixed, but I don't like how the video being not archived triggers the retry. Do we have to retry on 404?

Reply:
> Do we have to retry on 404?

wayback machine is a bit finicky and will sometimes return 404 even if it is archived, hence the original intent for retry. 

but someone suggested in the discord that they will cache it so retrying on this may not help anyways.

should keep it for other network errors, though.